app/controllers/admin/listing_shapes_controller.rb

```diff
@@ -78,6 +78,18 @@ def update
 
     if update_result.success
       flash[:notice] = t("admin.listing_shapes.edit.update_success", shape: pick_translation(shape[:name]))
+
+      # Onboarding wizard step recording
+      state_changed = Admin::OnboardingWizard.new(@current_community.id)
+        .update_from_event(:listing_shape_updated, update_result)
+      if state_changed
+        report_to_gtm([{event: "km_record", km_event: "Onboarding payments setup"},
+                       {event: "km_record", km_event: "Onboarding payment disabled"}])
+
+        with_feature(:onboarding_redesign_v1) do
+          flash[:show_onboarding_popup] = true
```

Comment:
To make the popup show this needs the redirect target action to check for this parameter and put the popup locals in place (using the view util). Plus then the view code needs to include the popup partial. The view code can be simply copy&pasted from some other view that shows the popup.